### ISSUE TITLE
Cannot execute operation because FlutterJNI is not attached to native

### ISSUE THREAD

**Issue by PreviousTlx** [flutter/flutter#63555]
## Steps to Reproduce

crash devices
```
vivo X21i.   android 9

vivo Y3.      android 9

HUAWEI畅玩 7C   android 8.0

GLK-AL00.    android 9

```

**Expected results:** 

**Actual results:** 

<details>
  <summary>Logs</summary>

```
java.lang.RuntimeException: Cannot execute operation because FlutterJNI is not attached to native.
	at io.flutter.embedding.engine.FlutterJNI.ensureAttachedToNative(FlutterJNI.java:222)
	at io.flutter.embedding.engine.FlutterJNI.setSemanticsEnabled(FlutterJNI.java:500)
	at io.flutter.embedding.engine.systemchannels.AccessibilityChannel.onAndroidAccessibilityEnabled(AccessibilityChannel.java:102)
	at io.flutter.view.AccessibilityBridge$2.onAccessibilityStateChanged(AccessibilityBridge.java:278)
	at android.view.accessibility.AccessibilityManager.lambda$notifyAccessibilityStateChanged$0(AccessibilityManager.java:1152)
	at android.view.accessibility.-$$Lambda$AccessibilityManager$yzw5NYY7_MfAQ9gLy3mVllchaXo.run(Unknown Source:4)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:224)
	at android.app.ActivityThread.main(ActivityThread.java:7042)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:536)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:868)
```

```
```

flutter doctor -v
```
[✓] Flutter (Channel stable, v1.17.5, on Mac OS X 10.14.5 18F132, locale
    yue-Hant-CN)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[!] Xcode - develop for iOS and macOS (Xcode 11.0)
    ! CocoaPods 1.7.2 out of date (1.8.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade:
        sudo gem install cocoapods
[✓] Android Studio (version 3.5)
[!] IntelliJ IDEA Ultimate Edition (version 2019.1.2)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.45.1)
[✓] Connected device (1 available)
```

</details>

**Comment by TahaTesser** [comment]
Hi @PreviousTlx 
Is this just flutter app or you using flutter into an existing app? 
Can you please provide  your `pubspec.yaml`, the list of affected devices and if possible your `flutter run --release --verbose` running on the same OS version as the affected devices.
Thank you

**Comment by PreviousTlx** [comment]
Hi @TahaTesser 
Using flutter into an existing app
、、、version: 1.0.0+1

environment:
  sdk: ">=2.1.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.2

dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
#  flutter_svg: ^0.17.4
#  flutter_html: ^1.0.0-pre.1
  package_info: ^0.4.0+18
  webview_flutter: ^0.3.21
  device_info: ^0.4.2+4
  flutter_easyrefresh: ^2.1.1
  dio: ^3.0.9
  connectivity: ^0.4.8+6
  cached_network_image: ^2.2.0+1
  fluttertoast: ^4.0.1
  flutter_swiper: ^1.1.6
  crypto: ^2.1.3
  flutter_staggered_grid_view: ^0.3.0
  chewie: ^0.9.10
  wakelock: ^0.1.4+2
  flutter_widget_from_html_core: ^0.3.3+4
  flutter_custom_dialog: ^1.0.20
  photo_view: ^0.9.2
  sliding_up_panel: ^1.0.2
#  flutter_ijkplayer: ^0.3.5+1
# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

flutter:
  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  assets:
    - assets/images/
、、、

**Comment by TahaTesser** [comment]
Since this is the production app, i am unable to reproduce it, seems to related to accessibility

**Comment by SebAubin** [comment]
I get this error:

``` [ERROR:flutter/shell/platform/android/platform_view_android_jni.cc(39)] java.lang.UnsatisfiedLinkError: Can't obtain class com.sun.jna.Pointer
08-13 14:35:59.628 20513 20513 E flutter : 	at com.sun.jna.Native.initIDs(Native Method)
08-13 14:35:59.628 20513 20513 E flutter : 	at com.sun.jna.Native.<clinit>()
08-13 14:35:59.628 20513 20513 E flutter : 	at com.sun.jna.Native.a()
08-13 14:35:59.628 20513 20513 E flutter : 	at h.a.a.d.<clinit>()
08-13 14:35:59.628 20513 20513 E flutter : 	at h.a.a.g.<init>()
08-13 14:35:59.628 20513 20513 E flutter : 	at net.devolutions.authenticator.MainActivity.d()
08-13 14:35:59.628 20513 20513 E flutter : 	at net.devolutions.authenticator.MainActivity$b.a()
08-13 14:35:59.628 20513 20513 E flutter : 	at e.a.c.a.k$a.a()
08-13 14:35:59.628 20513 20513 E flutter : 	at io.flutter.embedding.engine.b.b.a()
08-13 14:35:59.628 20513 20513 E flutter : 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage()
08-13 14:35:59.628 20513 20513 E flutter : 	at android.os.MessageQueue.nativePollOnce(Native Method)
08-13 14:35:59.628 20513 20513 E flutter : 	at android.os.MessageQueue.next(MessageQueue.java:323)
08-13 14:35:59.628 20513 20513 E flutter : 	at android.os.Looper.loop(Looper.java:136)
08-13 14:35:59.628 20513 20513 E flutter : 	at android.app.ActivityThread.main(ActivityThread.java:6682)
08-13 14:35:59.628 20513 20513 E flutter : 	at java.lang.reflect.Method.invoke(Native Method)
08-13 14:35:59.628 20513 20513 E flutter : 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
08-13 14:35:59.628 20513 20513 E flutter : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
08-13 14:35:59.628 20513 20513 E flutter :
08-13 14:35:59.628 20513 20513 F flutter : [FATAL:flutter/shell/platform/android/platform_view_android_jni.cc(76)] Check failed: CheckException(env).
08-13 14:35:59.629 20513 20513 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 20513 (s.authenticator)```

Seems to be related even though it's not EXACTLY the same error as you guys

**Comment by pengrad** [comment]
@PreviousTlx did you try latest stable 1.20.1 release?

**Comment by pengrad** [comment]
Hi @TahaTesser 
We also have this crash, this PR should fix it https://github.com/flutter/engine/pull/17311#issuecomment-674776686
I also tried to explain there why it happens with absolutely correct app setup.
Please consider to make a hotfix with it 🙏

**Comment by chinmaygarde** [comment]
Is there a reduced test case for this? I have asked @jason-simmons to take another look at the PR linked by @pengrad. The backtrace linked by @SebAubin seems like a platform channels call threw an exception and is a plugin issue unrelated to this one.

**Comment by SebAubin** [comment]
@chinmaygarde I'm pretty sure it's related since my app works when I don't use the flutter tool to compile the app.

**Comment by PreviousTlx** [comment]
> @PreviousTlx did you try latest stable 1.20.1 release?

It also happen in 1.20.1 stable。#33511

**Comment by BeshoyAdelHemaya** [comment]
Issue still happen in 1.20.1

**Comment by pengrad** [comment]
@PreviousTlx are you using `io.flutter.embedding.android.FlutterView` or legacy `io.flutter.view.FlutterView`?
I see new PR above, but it will fix only legacy and won't help in my case :(

**Comment by PreviousTlx** [comment]
> @PreviousTlx are you using `io.flutter.embedding.android.FlutterView` or legacy `io.flutter.view.FlutterView`?
> I see new PR above, but it will fix only legacy and won't help in my case :(

I am using io.flutter.embedding.android.FlutterView

**Comment by pengrad** [comment]
@chinmaygarde @TahaTesser @jason-simmons 
This issue should not be closed.
Linked PR https://github.com/flutter/engine/pull/20610 is fixing old legacy `io.flutter.view.FlutterView` but crash happens when using `io.flutter.embedding.android.FlutterView`

It is actually not related with any FlutterView and can be fixed in `AccessibilityBridge/Channel` level instead.
@jason-simmons could you give any feedback on my comment here? https://github.com/flutter/engine/pull/17311#issuecomment-674776686

This PR https://github.com/flutter/engine/pull/17311 breaks all tests in `AccessibilityBridgeTest` (I've tried locally, not sure why PR is green)
Would you accept PR patching `AccessibilityBridge` with additional tests?

I can reproduce exception by this test:
```java
  @Test
  public void itDoesNotCallChannelOnDetachedJNI() {
    FlutterJNI flutterJNI = new FlutterJNI();
    AccessibilityChannel channel = new AccessibilityChannel(mock(DartExecutor.class), flutterJNI);
    AccessibilityManager mockManager = mock(AccessibilityManager.class);
    when(mockManager.isEnabled()).thenReturn(true);

    // throws RuntimeException: FlutterJNI is not attached to native
    AccessibilityBridge bridge = setUpBridge(null, channel, mockManager, null, null, null);
    // this could be actual test after fix:
    verify(channel, never()).onAndroidAccessibilityEnabled();
  }
```
